packages/editor-ui/src/composables/usePushConnection.ts

```diff
@@ -493,6 +521,17 @@ export function usePushConnection({ router }: { router: ReturnType<typeof useRou
 		} else if (receivedData.type === 'nodeExecuteAfter') {
 			// A node finished to execute. Add its data
 			const pushData = receivedData.data;
+
+			/**
+			 * When we receive a placeholder in `nodeExecuteAfter`, we fake the items
+			 * to be the same count as the data the placeholder is standing in for.
+			 * This prevents the items count from jumping up when the execution
+			 * finishes and the full data replaces the placeholder.
+			 */
+			if (pushData.itemCount && pushData.data.data) {
+				pushData.data.data.main[0]?.push(...new Array(pushData.itemCount - 1).fill({ json: {} }));
```

Comment:
Shouldn't we fill this only if the data contains less than item count of items? At the moment if `itemCount` is set, the data will always be empty, but that's not apparent just by looking at this code. Checking the count would make this clearer and future-proof